docs/source/qvm.rst

```diff
@@ -36,43 +52,242 @@ QVM server by typing ``qvm -S`` into your terminal. You should see the following
     from pyquil.gates import *
-    qc = get_qc('9q-square-qvm')
 
+    qc = get_qc('9q-square-qvm')            # not general to any number of qubits, 9q-square-qvm is special
 
-One executes quantum programs on the QVM using the ``.run(...)`` method, intended to closely mirror how one will
-execute programs on a real QPU. We also offer a Wavefunction Simulator (formerly a part of the ``QVM`` object),
-which allows users to contruct and inspect wavefunctions of quantum programs. Learn more
-about the Wavefunction Simulator :ref:`here <wavefunction_simulator>`. For information on constructing quantum
-programs, please refer back to :ref:`basics`.
+    qubits = qc.qubits()                    # this information comes from qc.device
+    p = Program()
+    # ... build program, potentially making use of the qubits list
+
+    compiled_program = qc.compile(p)        # this makes multiple calls to qc.compiler
+
+    results = qc.run(compiled_program)      # this makes multiple calls to qc.qam
+
+.. note::
+
+    In addition to a running QVM server, you will need a running ``quilc`` server to compile your program. Setting
+    up both of these is very easy, as explained :ref:`here <server>`.
+
+
+The ``.run_and_measure(...)`` method
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This is the most high level way to run your program. With this method, you are **not** responsible for compiling your program
+before running it, nor do you have to specify any ``MEASURE`` instructions; all qubits will get measured.
+
+.. code:: python
+
+    from pyquil import Program, get_qc
+    from pyquil.gates import X
+
+    qc = get_qc("8q-qvm")
+
+    p = Program(X(0))
+
+    results = qc.run_and_measure(p, trials=5)
+    print(results)
+
+Trials specifies how many times to run this program. Let's see our results:
+
+.. parsed-literal::
+
+    {0: array([1, 1, 1, 1, 1]),
+     1: array([0, 0, 0, 0, 0]),
+     2: array([0, 0, 0, 0, 0]),
+     3: array([0, 0, 0, 0, 0]),
+     4: array([0, 0, 0, 0, 0]),
+     5: array([0, 0, 0, 0, 0]),
+     6: array([0, 0, 0, 0, 0]),
+     7: array([0, 0, 0, 0, 0])}
+
+The return value is a dictionary from qubit index to results for all trials.
+Every qubit in the lattice is measured for you, and as expected, qubit 0 has been flipped to the excited state
+for each trial.
 
 The ``.run(...)`` method
-------------------------
+^^^^^^^^^^^^^^^^^^^^^^^^
 
-The ``.run(...)`` method takes in a compiled program. You are responsible for compiling your program before running it.
-Remember to also start up a ``quilc`` compiler server, too, with ``quilc -S``.
+The lower-level ``.run(...)`` method gives you more control over how you want to build and compile your program than
+``.run_and_measure`` does. **You are responsible for compiling your program before running it.**
+The above program would be written in this way to execute with ``run``:
 
 .. code:: python
 
+    from pyquil import Program, get_qc
+    from pyquil.gates import X
```

Comment:
Change to `from pyquil.gates import X, MEASURE` since you're explicitly using `MEASURE` below (otherwise the code below would raise an error).

Reply:
Thank you!!!

Reply:
```suggestion
    from pyquil.gates import X, MEASURE
```